sync/sync-impl/src/main/java/com/duckduckgo/sync/impl/SyncRepository.kt

```diff
@@ -191,6 +204,30 @@ class AppSyncRepository @Inject constructor(
         return syncStore.token ?: ""
     }
 
+    override fun getConnectedDevices(): Result<List<ConnectedDevice>> {
+        val token = syncStore.token.takeUnless { it.isNullOrEmpty() }
```

Comment:
this is becoming a pattern, checking for a value in the store then returning an Error if it doesn't exist. How do you feel about moving all that logic to the store itself?

Reply:
Good point about being pattern (also in SyncApi and other parts). Probably something to extract, but it should stay somehow in the same class. That check is more related to the action the user is taking than to the value in the store. Inside the store, the empty token is a valid state, so I wouldn't move it there.

However, I don't know if we will need that level of error granularity in the final UI, so probably most of them will be removed.